DiztinGUIsh/static/LogCreator.cs

```diff
@@ -467,7 +519,23 @@ private static string GetXFlag(int offset, int length)
         // output label at snes offset, and its value
         private static string GetLabelAssign(int offset, int length)
         {
-            string s = string.Format("{0} = {1}", Data.GetLabel(offset), Util.NumberToBaseString(offset, Util.NumberBase.Hexadecimal, 6, true));
+            var labelName = Data.GetLabelName(offset);
+            var offsetStr = Util.NumberToBaseString(offset, Util.NumberBase.Hexadecimal, 6, true);
+            var labelComment = Data.GetLabelComment(offset);
+
+            if (string.IsNullOrEmpty(labelName))
+                return "";
+
+            if (labelComment == null)
+                labelComment = "";
+
+            var finalCommentText = "";
+
+            // TODO: sorry, probably not the best way to stuff this in here, consider putting it in the %comment% section in the future. -Dom
+            if (printLabelSpecificComments && labelComment != "")
+                finalCommentText = $"; !^ {labelComment} ^!";
+
+            string s = $"{labelName} = {offsetStr}{finalCommentText}";
```

Comment:
this is a bit of a mess in terms of formatting, but, for normal people they won't have this option checked.  I wanted to add some characters to make it easy to later parse out labels and comments from the asm.